reticulator/reticulator.py

```diff
@@ -341,18 +341,26 @@ def __str__(self):
         return json.dumps(self.data, indent=2, ensure_ascii=False)
 
 
+    def jsonpath_exists(self, json_path:str) -> bool:
+        """
+        Checks if a jsonpath exists
+        """
+        try:
+            self.get_jsonpath(json_path)
+            return True
+        except AssetNotFoundError:
+            return False
+
     def delete_jsonpath(self, json_path:str, ensure_exists:bool = False) -> None:
         """
         Removes value at jsonpath location.
         """
-        try:
+        path_exists = self.jsonpath_exists
+        if path_exists:
             dpath.util.delete(self.data, json_path)
-        except dpath.exceptions.PathNotFound as path_not_found:
+        else:
```

Comment:
This else: if can be rewritten as an `elif`, which will keep the code less indented.